edgedb/protocol/protocol.pyx

```diff
@@ -480,14 +476,41 @@ cdef class SansIOProtocol:
         inline_typenames: bool = False,
         inline_typeids: bool = False,
         allow_capabilities: enums.Capability = enums.Capability.ALL,
+        state: typing.Optional[dict] = None,
     ):
         cdef:
             BaseCodec in_dc
             BaseCodec out_dc
+            WriteBuffer buf
 
         self.ensure_connected()
         self.reset_status()
 
+        if state is not None:
+            if self.state_codec is None:
+                try:
+                    await self._parse(
+                        "select 0",
```

Comment:
Rather than forcing the clients to do hacks like this, why can't we push the initial `StateDataDescription` message on connection?